File: C/12_Sort/BOJ_03_10989_수정렬하기-3.c
#include <stdio.h>
main() {
	int n, num;
	int cnt[10001] = { 0 }, sum[10001] = { 0 };
	scanf("%d", &n);
	for (int i = 0; i < n; i++) {
		scanf("%d", &num);
		cnt[num]++;
	}
	sum[0] = cnt[0];
	for (int i = 1; i < 10001; i++) {
		sum[i] = sum[i - 1] + cnt[i];
	}
	for (int j = 1; j <= 10001; j++) {
		if (cnt[j - 1] != 0) {
			if (j != 1) {
				for (int k = 0; k < sum[j - 1] - sum[j - 2]; k++) {
					printf("%d\n", j - 1);
				}
			}
			else {
				for (int k = 0; k < sum[0]; k++) {
					printf("%d\n", 0);
				}
			}
		}
		if (sum[j - 1] == n) break;
	}
}

File: JAVA/12_Sort/BOJ_03_10989_수정렬하기-3.java
import java.io.IOException;
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.BufferedWriter;
import java.io.OutputStreamWriter;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        int n = Integer.parseInt(br.readLine());
        int[] cnt = new int[10001];
        for (int i = 0; i < n; i++) {
            cnt[Integer.parseInt(br.readLine())]++;
        }
        int sum = 0;
        for (int i = 1; i <= 10000; i++) {
            bw.write(String.valueOf(i + "\n").repeat(cnt[i]));
            if (cnt[i] != 0) {
                for (int j = 0; j < cnt[i]; j++) {
                    sum++;
                }
            }
            if (sum == n) break;
        }
        bw.flush();
        bw.close();
    }
}